src/components/Heading/Heading.styles.ts

```diff
@@ -0,0 +1,23 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { fontWeights } from '../../util/styles/fonts';
+import { mergeStyleSets } from '@uifabric/styling';
+import { memoizeFunction } from '@uifabric/utilities';
+import { HeadingSize } from './Heading';
+
+export interface HeadingStyleProps {
+  size?: HeadingSize;
+}
+
+export const getClassNames = memoizeFunction((styleProps: HeadingStyleProps) => {
+  const { size } = styleProps;
+
+  return mergeStyleSets({
+    root: {
+      display: 'inline',
+      fontSize: 'inherit',
+      lineHeight: 'inherit',
+      fontWeight: size === '4' ? fontWeights.normal : size === 'none' ? 'inherit' : fontWeights.bold,
```

Comment:
i'm surprised tslint allows nested ternaries, they are pretty hard to read. can we break this up?

Reply:
TBH, they're kind of growing on me. I had it split up with parens a bit more, but prettier stripped them out.
I've broken it up now.